transcom/mymove
# B 20473 office user cancel move

## [Agility ticket](https://www13.v1host.com/USTRANSCOM38/story.mvc/Summary?oidToken=Story%3A980089&RoomContext=TeamRoom%3A848240&concept=TeamRoom)

## [INT PR](https://github.com/transcom/mymove/pull/13957)
## [INT PR 2](https://github.com/transcom/mymove/pull/14090)

## Summary
This story will add the ability for the SC and TOO to Cancel a move from the Move Details page.  The move cannot have shipment in Approved status (non-PPM) or in Closeout complete status (PPM).  Also, move cannot be in Canceled status.

### How to test

1. Access the SC Counseling queue.  Select a move in Needs Services Counseling status.
2. Click on the Cancel move link on the top of the Move details page.
3. Click on the Cancel Move button on the confirmation modal.
4. Verify that the Cancel link is now removed and that the shipment cards now display Canceled.
5. Navigate back to the SC queue, verify that the move is no longer in the queue.
6. Navigate to the Search tab and search for the move that was canceled.
7. Verify that it has a Canceled status.
8. Access the TOO Task Order queue, run through steps 2-7 for a move that does not have any previously approved shipments.
9. Go back as a TOO and find a move that does have at least 1 approved shipment, verify that the Cancel move link is not available.
10. Find a shipment in Closeout Complete status and verify for that move that the Cancel move link is not available.
